pilot/pkg/features/pilot.go

```diff
@@ -221,12 +221,17 @@ var (
 			"if headless services have a large number of pods.",
 	).Get()
 
-	EnableRemoteJwks = env.Register(
-		"PILOT_JWT_ENABLE_REMOTE_JWKS",
-		false,
-		"If enabled, checks to see if the configured JwksUri in RequestAuthentication is a mesh cluster URL "+
-			"and configures remote Jwks to let Envoy fetch the Jwks instead of Istiod.",
-	).Get()
+	JwksFetchMode = func() jwt.JwksFetchMode {
+		v := env.Register(
+			"PILOT_JWT_ENABLE_REMOTE_JWKS",
```

Comment:
Seems thats what we trying to avoid.

Reply:
Why? We can introduce a new environment variable, and mark the old as deprecated and document it in the release notes. The new mesh deployment will use the new environment variable. The code will handle both new and old environment variables for a few releases, and eventually remove the old env var. That's the typical way of deprecating API.

Reply:
I would prefer to use the existing one. Why add churn for users when its not needed? I don't see any benefits

Reply:
The value is quite confusing. "istiod==false", "hybrid==true". Unless the user understands the history of the code, it is hard to tell which value to set here.

Reply:
Users should only set `istiod`, `hybrid`, or `envoy`. We can document only those fields if its confusing. You can migrate users off `true` and `false` without breaking all users that *don't*.

Reply:
Do we still call it "ENABLE REMOTE JWKS", with the new values? Will it be confusing? Instead of migrating to new values with old variable, why don't we migrate to the new variable? The effort of migration is similar (migrating values or migrating variable), and the end result will be cleaner if we use a new env variable.

Reply:
This is internal break-glass advanced user feature, we have far worse names than this one. Its not some public facing API. Optimizing for "clean" is not worthwhile IMO

Reply:
Environment variable is also public API. Since we decide to change the values for this environment variable, which requires documentation and migration, I would rather do it with a new variable.

Reply:
Marco mentioned that we do not plan to deprecate the old values ("true" and "false") for backwards compatibility. That is fine. We can keep the current variable in this case.